### ISSUE TITLE
InstagramLoader doesn't load images

### ISSUE THREAD

**Issue by korakinos** [keshrath/ImageLoader#1]
InstagramLoader isn't working for me: The resulting image list appears to be always empty. This is the code I am using (modified from the example code by adding debug printlns, excluding my access token and client id):

```
import at.mukprojects.imageloader.*;
import at.mukprojects.imageloader.instagram.*;
import at.mukprojects.imageloader.image.*;

// API with access token
String accessToken = /*my access token*/;

// API with client ID
String clientId = /*my client id*/;

ImageLoader loader;
ImageList list;
Image img;

void setup() {
  size(800, 450);

  // API with access token
  loader = new InstagramLoader(this, accessToken, "");

  // API with client ID
   //loader = new InstagramLoader(this, clientId);

  list = loader.start("vienna", false, 60 * 1000);
  while (list.size() == 0) {
    println("No images!");
    delay(1000);
  }
}

void draw() {
  if (img == null) {
    println("Picking random image...");
    img = list.getRandom();
    if (img != null)
      println("Image picked.");
  } else {
    image(img.getImg(), 0, 0, width, height);
  }
}

void mousePressed() {
  img = list.getRandom();
}
```

When I run it, it is stuck in setup(), outputting "No images!" continuously. Other than that, no error message.

(Edit: I just tested the FlickrLoader and it works, so it must really be about InstagramLoader.)

**Comment by korakinos** [comment]
Sorry, if I am not mistaken, I just realised how restrictive exactly the "sandbox mode" of Instagram is: A sandboxed app cannot access any images outside the sandbox, and since my current sandbox has no users besides me and I haven't uploaded anything yet, of course the above code doesn't load any images… (I cannot confirm this at the moment as I have no way of uploading, but will do soon.)

**Comment by keshrath** [comment]
Hi,
It seems that this is the problem. I tested your code with my key and it loads the images. But there remains an error with the URL parser. It seems that they have change the API slightly and added a cache id. Also I'm pretty sure that it had worked in sandbox mode so they must have changes that as well. I will fix the parser tomorrow and upload a new version. Please keep me updated if the sandbox mode was the problem.

**Comment by korakinos** [comment]
It works now. There were actually two sources of error: First I needed to have a _tagged_ image or video in my app's sandbox for InstagramLoader.start() to find. Second, it wouldn't recognise the file format ("Could not find a method to load […]", an error that I recognize from the underlying JInstagram library, which I am using directly now for my project, btw), leading to a NullPointerException in the test sketch above. – But that you already fixed in version 3.0.0 or 3.0.1 of ImageLoader, so all is good now. Thanks!
